### ISSUE TITLE
Mixed Precision fails Assertion Error in Gradient Scaling

### ISSUE THREAD

**Issue by j93hahn** [nerfstudio-project/nerfstudio#1694]
**Describe the bug**
Training Nerfacto on multiple scenes with mixed precision fails to pass an assertion error (detailed below). However, if I try to train the model without mixed precision, the convergence plot is very bad and it's evident that the model does not learn as well as it would have been with mixed precision.

**To Reproduce**
1. Clone a fresh checkout of Nerfstudio library
2. Run this command to train Nerfacto on Lego blender using mixed precision: `python /path/to/nerfstudio/scripts/train.py nerfacto --vis wandb --data="/path/to/data/for/blender/lego/" --experiment-name="blender_lego" --relative-model-dir=nerfstudio_models/ --steps-per-save=0 --logging.local-writer.enable=False --logging.enable-profiler=False --pipeline.model.background-color white --pipeline.model.proposal-initial-sampler uniform --pipeline.model.near-plane 2. --pipeline.model.far-plane 6. --pipeline.datamanager.camera-optimizer.mode off --pipeline.model.use-average-appearance-embedding False --pipeline.model.distortion-loss-mult 0 --pipeline.model.disable-scene-contraction True blender-data` --> this command produces the bug
3. Run this command to train Nerfacto on Lego blender without using mixed precision: `python /path/to/nerfstudio/scripts/train.py nerfacto --vis wandb --data="/path/to/data/for/blender/lego/" --experiment-name="blender_lego" --relative-model-dir=nerfstudio_models/ --steps-per-save=0 --logging.local-writer.enable=False --logging.enable-profiler=False --pipeline.model.background-color white --pipeline.model.proposal-initial-sampler uniform --pipeline.model.near-plane 2. --pipeline.model.far-plane 6. --pipeline.datamanager.camera-optimizer.mode off --pipeline.model.use-average-appearance-embedding False --pipeline.model.distortion-loss-mult 0 --pipeline.model.disable-scene-contraction True --mixed-precision=False blender-data` --> this command gets rid of the bug but does very bad in training

**Expected behavior**
When I train the Nerfacto model with mixed precision, it produces this error:
```
Traceback (most recent call last):
  File "/share/data/pals/jjahn/temp/nerfstudio/scripts/train.py", line 256, in <module>
    entrypoint()
  File "/share/data/pals/jjahn/temp/nerfstudio/scripts/train.py", line 247, in entrypoint
    main(
  File "/share/data/pals/jjahn/temp/nerfstudio/scripts/train.py", line 233, in main
    launch(
  File "/share/data/pals/jjahn/temp/nerfstudio/scripts/train.py", line 172, in launch
    main_func(local_rank=0, world_size=world_size, config=config)
  File "/share/data/pals/jjahn/temp/nerfstudio/scripts/train.py", line 87, in train_loop
    trainer.train()
  File "/share/data/pals/jjahn/nerfstudio/nerfstudio/engine/trainer.py", line 218, in train
    loss, loss_dict, metrics_dict = self.train_iteration(step)
  File "/share/data/pals/jjahn/nerfstudio/nerfstudio/utils/profiler.py", line 43, in wrapper
    ret = func(*args, **kwargs)
  File "/share/data/pals/jjahn/nerfstudio/nerfstudio/engine/trainer.py", line 395, in train_iteration
    self.optimizers.optimizer_scaler_step_all(self.grad_scaler)
  File "/share/data/pals/jjahn/nerfstudio/nerfstudio/engine/optimizers.py", line 130, in optimizer_scaler_step_all
    grad_scaler.step(optimizer)
  File "/share/data/pals/jjahn/mc3/envs/alpha/lib/python3.10/site-packages/torch/cuda/amp/grad_scaler.py", line 368, in step
    assert len(optimizer_state["found_inf_per_device"]) > 0, "No inf checks were recorded for this optimizer."
AssertionError: No inf checks were recorded for this optimizer.
```

I have run this exact command (with mixed precision) many times in the past and it's always worked completely fine (hits eval PSNRs of 33 which is expected for lego). But today, I tried to train Nerfacto on a different dataset and it just broke down because of this bug. I looked it up here: https://discuss.pytorch.org/t/no-inf-checks-were-recorded-for-this-optimizer/140505 which says that the error lies from the user not calling `backward()` on the scaled loss. But I see that it does get called here in the Trainer pipeline: https://github.com/nerfstudio-project/nerfstudio/blob/f6b67a0ad0d3fe843c41711825e40da5d99f9f85/nerfstudio/engine/trainer.py#L395 so that can't be the issue.

Does anyone know why using mixed precision causes this error to occur? It has never happened to me until today which is why I am extra confused. I am using PyTorch 2.0.0 with CUDA 11.8 and CUDA driver 525.89.02; could this be the error?

**Comment by ksnzh** [comment]
for this moment, dont use pytorch 2.0. Just use pytorch 1.13.1 pre-built with cuda 11.7 and it's  compatible.

**Comment by jkulhanek** [comment]
PyTorch 2.0 is now supported. The error should be resolved #1919
